### PR TITLE
Converted Constants into enum and added LineNo to Error Message

### PR DESCRIPTION
**Pull request by SpocWeb**
As described in Gitter.

### CODE REVIEW

File: src/Core/CodeModel/CSharpExpressionWriter.cs [hunk 1]
```diff
@@ -16,84 +16,76 @@
...
-            { CodeOperatorType.ModEq, 1 },
-            { CodeOperatorType.MulEq, 1 },
-            { CodeOperatorType.AndEq, 1 },
-            { CodeOperatorType.OrEq, 1 },
-            { CodeOperatorType.DivEq, 1 },
-            { CodeOperatorType.ShlEq, PrecAssignment },
-            { CodeOperatorType.ShrEq, PrecAssignment },
-            { CodeOperatorType.XorEq, PrecAssignment },
+
+        enum Prec
```
Comment (uxmal) [inline src/Core/CodeModel/CSharpExpressionWriter.cs]: Let's call this `Precedence`.

File: src/Core/Resources.resx [hunk 2]
```diff
@@ -153,7 +153,7 @@
     <value>Failed to create tmp directory {0}.</value>
   </data>
   <data name="ErrInvalidCharacter" xml:space="preserve">
-    <value>Invalid character '{0}' (U+{1:X4}).</value>
+    <value>Invalid character '{0}' (U+{1:X4}) in Line {2}.</value>
```
Comment (uxmal) [inline src/Core/Resources.resx]: Change to " in line " (lower case).

File: src/Core/CodeModel/CSharpExpressionWriter.cs [hunk 3]
```diff
@@ -16,84 +16,76 @@
...
+            Shift = 10,
+            Additive = 11,
+            Multplicative = 12,
+            Unary = 13,
+            Postfix = 14,
+            Primary = 15,
+
+        }
+
+        private static Prec operatorPrecedence(CodeOperatorType op) => op switch
```
Comment (uxmal) [inline src/Core/CodeModel/CSharpExpressionWriter.cs]: I'm curious why you changed this from a Dictionary lookup?

File: src/Core/CodeModel/CSharpExpressionWriter.cs (hunk 1 above)
Comment (SpocWeb) [reply src/Core/CodeModel/CSharpExpressionWriter.cs]: Very good, I wanted to keep the change minimal, but the full name is better. 
What do you think of removing the Number Values? They are superfluous and a possible source of error.

File: src/Core/Resources.resx (hunk 2 above)
Comment (SpocWeb) [reply src/Core/Resources.resx]: Done.

File: src/Core/CodeModel/CSharpExpressionWriter.cs (hunk 3 above)
Comment (SpocWeb) [reply src/Core/CodeModel/CSharpExpressionWriter.cs]: t is slightly faster and easier to maintain should the precedences ever need to be amended. 
It looks more concise and does the job. Aesthetics, sorry.

**Comment by uxmal** [comment]
Looks good; thanks for the contribution!

**Pull request merged**